### PR TITLE
Add default resizer and default resizer adapter

### PR DESCRIPTION
**Pull request by hanovruslan**
Implementation of #700

### CODE REVIEW

**Comment by hanovruslan** [comment]
ping @rande

**Comment by hanovruslan** [comment]
ping @rande  )

**Comment by hanovruslan** [comment]
@rande i see other PRs have response, and why this is not ? Is it really not good for bundle ?

**Comment by hanovruslan** [comment]
Anybody ping?)

**Comment by hanovruslan** [comment]
ping @rande )

File: Tests/DependencyInjection/SonataMediaExtensionTest.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            $this->isInstanceOf('Imagine\\Gd\\Imagine', $this->container->get('sonata.media.adapter.image.default'));
+            $this->isInstanceOf('Imagine\\Gd\\Imagine', $this->container->get('sonata.media.adapter.image.gd'));
+        }
+        if (extension_loaded('imagick')) {
+            $this->isInstanceOf('Imagine\\Imagick\\Imagine', $this->container->get('sonata.media.adapter.image.imagick'));
+        }
+        if (extension_loaded('gmagick')) {
+            $this->isInstanceOf('Imagine\\Gmagick\\Imagine', $this->container->get('sonata.media.adapter.image.gmagick'));
+        }
+        if (!extension_loaded('gd') && !extension_loaded('imagick') && extension_loaded('gmagick')) {
```
Comment (Soullivaneuh) [inline Tests/DependencyInjection/SonataMediaExtensionTest.php]: Should be `!extension_loaded('gmagick')` I think.

File: Tests/DependencyInjection/SonataMediaExtensionTest.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            throw new RuntimeException('install at least one image adapter (gd, imagick or gmagick)');
```
Comment (Soullivaneuh) [inline Tests/DependencyInjection/SonataMediaExtensionTest.php]: At least, I don't think test should fail if no extension loaded, but should be skipped.

Maybe a refactor had to be done here (e.g. on test per extension).

File: Resizer/AdapterAwareTrait.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<?php
+
```
Comment (Soullivaneuh) [inline Resizer/AdapterAwareTrait.php]: Missing sonata header.

File: Resizer/AdapterAwareTrait.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+namespace Sonata\MediaBundle\Resizer;
+
+use Imagine\Image\ImagineInterface;
+
+trait AdapterAwareTrait
```
Comment (Soullivaneuh) [inline Resizer/AdapterAwareTrait.php]: Why using trait?

File: Tests/DependencyInjection/ConfigurationTest.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
+<?php
+
```
Comment (Soullivaneuh) [inline Tests/DependencyInjection/ConfigurationTest.php]: Missing Sonata headers.

File: Tests/DependencyInjection/SonataMediaExtensionTest.php [hunk 6]
```diff
@@ -0,0 +1,122 @@
+<?php
+
```
Comment (Soullivaneuh) [inline Tests/DependencyInjection/SonataMediaExtensionTest.php]: Missing Sonata headers.

**Comment by Soullivaneuh** [comment]
I'm not an active user of this bundle, but I'll try to take a look soon.

Can you please fix my quick review?

**Comment by hanovruslan** [comment]
@Soullivaneuh yep, i'll do it but later this day
and thanks for response !)

File: Resizer/AdapterAwareTrait.php (hunk 4 above)
Comment (hanovruslan) [inline Resizer/AdapterAwareTrait.php]: For the reason that adapter and its method-accesors using in two classes at least

File: Tests/DependencyInjection/SonataMediaExtensionTest.php (hunk 6 above)
Comment (hanovruslan) [inline Tests/DependencyInjection/SonataMediaExtensionTest.php]: Fixed

File: Tests/DependencyInjection/ConfigurationTest.php (hunk 5 above)
Comment (hanovruslan) [inline Tests/DependencyInjection/ConfigurationTest.php]: Fixed

File: Resizer/AdapterAwareTrait.php (hunk 3 above)
Comment (hanovruslan) [inline Resizer/AdapterAwareTrait.php]: Fixed

**Comment by hanovruslan** [comment]
How to pass "SonataCI - Quick Check" ? Did i missed smth in contributing rules?

**Comment by rande** [comment]
@hanovruslan I need to fix " SonataCi / Sandbox / PHPUnit — None ", there is nothing wrong on your side.

**Comment by hanovruslan** [comment]
```
there is nothing wrong on your side
```
Sounds good!

**Comment by Soullivaneuh** [comment]
Travis is still failing.

**Comment by hanovruslan** [comment]
Yep, i must learn that sonata has php 5.3 support. Sorry for that

File: Resizer/AdapterAwareTrait.php (hunk 4 above)
Comment (hanovruslan) [inline Resizer/AdapterAwareTrait.php]: Will be removed in order to support php 5.3

**Comment by hanovruslan** [comment]
```
Travis is still failing.
```
Fixed

**Comment by OskarStark** [comment]
ping @rande can you please review this PR?

thank you

File: DependencyInjection/Configuration.php [hunk 7]
```diff
@@ -457,6 +458,29 @@ private function addResizerSection(ArrayNodeDefinition $node)
...
+                    ->children()
+                        ->scalarNode('default')->defaultValue('sonata.media.resizer.simple')->end()
+                    ->end()
+                ->end()
+            ->end()
+        ;
+    }
+
+    /**
+     * @param \Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition $node
```
Comment (Soullivaneuh) [inline DependencyInjection/Configuration.php]: Don't use FQN. Import class instead.

File: DependencyInjection/SonataMediaExtension.php [hunk 8]
```diff
@@ -129,9 +130,44 @@ public function load(array $configs, ContainerBuilder $container)
         $this->configureExtra($container, $config);
         $this->configureBuzz($container, $config);
         $this->configureProviders($container, $config);
+        $this->configureAdapters($container, $config);
+        $this->configureResizers($container, $config);
         $this->configureClassesToCompile();
     }
```
Comment (Soullivaneuh) [inline DependencyInjection/SonataMediaExtension.php]: Missing doc block.

File: Resizer/ResizerInterface.php [hunk 9]
```diff
@@ -32,4 +33,14 @@ public function resize(MediaInterface $media, File $in, File $out, $format, arra
      * @return \Imagine\Image\Box
      */
     public function getBox(MediaInterface $media, array $settings);
+
+    /**
+     * @return \Imagine\Image\ImagineInterface
```
Comment (Soullivaneuh) [inline Resizer/ResizerInterface.php]: Don't use FQN. Import class instead.

File: Resizer/ResizerInterface.php [hunk 10]
```diff
@@ -32,4 +33,14 @@ public function resize(MediaInterface $media, File $in, File $out, $format, arra
...
+     */
+    public function getAdapter();
+
+    /**
+     * @param \Imagine\Image\ImagineInterface $adapter
```
Comment (Soullivaneuh) [inline Resizer/ResizerInterface.php]: Don't use FQN. Import class instead.

File: Resizer/SimpleResizer.php [hunk 11]
```diff
@@ -21,7 +21,25 @@
 
 class SimpleResizer implements ResizerInterface
 {
+    /**
+     * \Imagine\Image\ImagineInterface.
```
Comment (Soullivaneuh) [inline Resizer/SimpleResizer.php]: * `@var`
* Don't use FQN. Import class instead.

File: Resizer/SimpleResizer.php [hunk 12]
```diff
@@ -21,7 +21,25 @@
...
+     */
     protected $adapter;
+    /**
```
Comment (Soullivaneuh) [inline Resizer/SimpleResizer.php]: Need a newline separation.

File: Resizer/SimpleResizer.php [hunk 13]
```diff
@@ -21,7 +21,25 @@
...
+     * @return \Imagine\Image\ImagineInterface
```
Comment (Soullivaneuh) [inline Resizer/SimpleResizer.php]: Don't use FQN. Import class instead.

File: Resizer/SimpleResizer.php [hunk 14]
```diff
@@ -21,7 +21,25 @@
...
+     */
+    public function getAdapter()
+    {
+        return $this->adapter;
+    }
+
+    /**
+     * @param \Imagine\Image\ImagineInterface $adapter
```
Comment (Soullivaneuh) [inline Resizer/SimpleResizer.php]: Don't use FQN. Import class instead.

File: Resizer/SquareResizer.php [hunk 15]
```diff
@@ -29,9 +29,24 @@
 class SquareResizer implements ResizerInterface
 {
     /**
-     * ImagineInterface.
+     * \Imagine\Image\ImagineInterface.
```
Comment (Soullivaneuh) [inline Resizer/SquareResizer.php]: Don't use FQN. Import class instead.

File: Resizer/SquareResizer.php [hunk 16]
```diff
@@ -29,9 +29,24 @@
...
      */
     protected $adapter;
+    /**
```
Comment (Soullivaneuh) [inline Resizer/SquareResizer.php]: Need a new line separation.

File: Resizer/SquareResizer.php [hunk 17]
```diff
@@ -29,9 +29,24 @@
...
+     * @return \Imagine\Image\ImagineInterface
```
Comment (Soullivaneuh) [inline Resizer/SquareResizer.php]: Don't use FQN. Import class instead.

File: Resizer/SquareResizer.php [hunk 18]
```diff
@@ -29,9 +29,24 @@
...
+     */
+    public function getAdapter()
+    {
+        return $this->adapter;
+    }
+
+    /**
+     * @param \Imagine\Image\ImagineInterface $adapter
```
Comment (Soullivaneuh) [inline Resizer/SquareResizer.php]: Don't use FQN. Import class instead.

File: Tests/DependencyInjection/SonataMediaExtensionTest.php [hunk 19]
```diff
@@ -0,0 +1,153 @@
...
+namespace Sonata\MediaBundle\Tests\DependencyInjection;
+
+use PHPUnit_Framework_TestCase;
+use Sonata\MediaBundle\DependencyInjection\SonataMediaExtension;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class SonataMediaExtensionTest extends PHPUnit_Framework_TestCase
+{
+    /**
+     * @var \Sonata\MediaBundle\DependencyInjection\SonataMediaExtension
```
Comment (Soullivaneuh) [inline Tests/DependencyInjection/SonataMediaExtensionTest.php]: Don't use FQN. Import class instead.

File: Tests/DependencyInjection/SonataMediaExtensionTest.php [hunk 20]
```diff
@@ -0,0 +1,153 @@
...
+     */
+    private $extension;
+
+    /**
+     * Root name of the configuration.
+     *
+     * @var \Symfony\Component\DependencyInjection\ContainerBuilder
```
Comment (Soullivaneuh) [inline Tests/DependencyInjection/SonataMediaExtensionTest.php]: Don't use FQN. Import class instead.

File: Tests/DependencyInjection/SonataMediaExtensionTest.php [hunk 21]
```diff
@@ -0,0 +1,153 @@
...
+     */
+    private $container;
+
+    /**
+     * @return \Sonata\MediaBundle\DependencyInjection\SonataMediaExtension
```
Comment (Soullivaneuh) [inline Tests/DependencyInjection/SonataMediaExtensionTest.php]: Don't use FQN. Import class instead.

File: DependencyInjection/SonataMediaExtension.php (hunk 8 above)
Comment (hanovruslan) [inline DependencyInjection/SonataMediaExtension.php]: Fixed

File: Tests/DependencyInjection/SonataMediaExtensionTest.php (hunk 20 above)
Comment (hanovruslan) [inline Tests/DependencyInjection/SonataMediaExtensionTest.php]: Fixed

**Comment by Soullivaneuh** [comment]
Please squash your commits.

I let @rande review it.

**Comment by hanovruslan** [comment]
@Soullivaneuh done!

**Comment by hanovruslan** [comment]
... Ping? )

**Comment by OskarStark** [comment]
@hanovruslan can you please rebase your PR?

**Comment by hanovruslan** [comment]
I've removed adapter accessors from resizer types

**Comment by hanovruslan** [comment]
@OskarStark ci tests output something weird. can you please check it and tell me what should i do ?

**Comment by hanovruslan** [comment]
@SonataCI 

I've faced some conflicts after rebase and dont know how to resolve them properly

```
  Problem 1
    - Installation request for sonata-project/media-bundle 3.x-dev -> satisfiable by sonata-project/media-bundle[3.x-dev].
    - sonata-project/block-bundle 2.3.8 conflicts with sonata-project/media-bundle[3.x-dev].
    - Installation request for sonata-project/block-bundle 2.3.8 -> satisfiable by sonata-project/block-bundle[2.3.8]
```

/cc @OskarStark

**Comment by core23** [comment]
@hanovruslan SonataCI is just a bot, you can't ping him ;)

**Comment by core23** [comment]
We're updating all version, please be patient 

Refs https://github.com/sonata-project/SonataAdminBundle/issues/3731

**Comment by hanovruslan** [comment]
@core23 Yep, i know that but maybe github has some feature like mail groups ))

**Comment by hanovruslan** [comment]
```
We're updating all version, please be patient

Refs sonata-project/SonataAdminBundle#3731
```

@core23 Should i just wait for something ? As you can see this PR does not play with deps.

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

**Comment by OskarStark** [comment]
is this a BC break?

**Comment by hanovruslan** [comment]
@OskarStark as far as i know - no, there is no BC break. But. I've done this PR long time ago (more than year) and i think test on real app would be helpful

**Comment by OskarStark** [comment]
> i think test on real app would be helpful

are you willing to to this?

**Comment by hanovruslan** [comment]
@OskarStark yep, but in about a week i guess. Not today for sure

**Comment by OskarStark** [comment]
> @OskarStark yep, but in about a week i guess. Not today for sure

thanks for your effort 👍

File: Resizer/SimpleResizer.php [hunk 22]
```diff
@@ -22,7 +22,7 @@
 class SimpleResizer implements ResizerInterface
 {
     /**
-     * @var ImagineInterface
+     * @var ImagineInterface.
```
Comment (core23) [inline Resizer/SimpleResizer.php]: Please remove the `.`

File: Resizer/SquareResizer.php [hunk 23]
```diff
@@ -29,12 +29,28 @@
 class SquareResizer implements ResizerInterface
 {
     /**
-     * ImagineInterface.
+     * @var ImagineInterface.
```
Comment (core23) [inline Resizer/SquareResizer.php]: Please remove the `.`

**Comment by core23** [comment]
Please rebase and apply the new PR template @hanovruslan

File: DependencyInjection/SonataMediaExtension.php [hunk 24]
```diff
@@ -127,14 +128,57 @@ public function load(array $configs, ContainerBuilder $container)
...
 
     /**
      * @param ContainerBuilder $container
      * @param array            $config
      */
-    public function configureProviders(ContainerBuilder $container, $config)
+    public function configureAdapters(ContainerBuilder $container, array $config)
```
Comment (core23) [inline DependencyInjection/SonataMediaExtension.php]: Why public? Can we make this at least protected?

File: DependencyInjection/SonataMediaExtension.php [hunk 25]
```diff
@@ -127,14 +128,57 @@ public function load(array $configs, ContainerBuilder $container)
...
+            }
+        }
+        $container->setAlias('sonata.media.adapter.image.default', $config['adapters']['default']);
+    }
+
+    /**
+     * @param ContainerBuilder $container
+     * @param array            $config
+     */
+    public function configureResizers(ContainerBuilder $container, array $config)
```
Comment (core23) [inline DependencyInjection/SonataMediaExtension.php]: Same here. Why public?

File: Resizer/ResizerInterface.php [hunk 26]
```diff
@@ -12,6 +12,7 @@
 namespace Sonata\MediaBundle\Resizer;
 
 use Gaufrette\File;
+use Imagine\Image\ImagineInterface;
```
Comment (core23) [inline Resizer/ResizerInterface.php]: Not related to this PR

**Comment by Soullivaneuh** [comment]
> is this a BC break?

--

> @OskarStark as far as i know - no, there is no BC break. But. I've done this PR long time ago (more than year) and i think test on real app would be helpful

@hanovruslan in this case, please reopen this PR against the stable branch (3.x).

Thanks again. :+1: 

Closing this one.

**Pull request closed without merging**